### PR TITLE
profile api

### PR DESCRIPTION
**Pull request by ykssheng**
profile api

### CODE REVIEW

File: app/controllers/ProfileControllers.js [hunk 1]
```diff
@@ -178,4 +178,83 @@ exports.getProfileAccount = function (req, res, next, ProfileAccountID) {
...
+    // create a new variable to hold the article that was placed on the req object.
+    var profile = req.profile;
+
+    profile.updateAttributes({
+        FirstName: req.body.first_name,
+        LastName: req.body.last_name,
+        Email: req.body.email,
+        ContactNo: req.body.contact_no,
+        Gender: req.body.gender,
+        DOB: req.body.date_of_birth
```
Comment (jingwong25) [inline app/controllers/ProfileControllers.js]: use camel case after req.body.

File: app/controllers/ProfileControllers.js [hunk 2]
```diff
@@ -178,4 +178,83 @@ exports.getProfileAccount = function (req, res, next, ProfileAccountID) {
...
+/**
+ * New version 2.0
+ */
+
+/**
+ * Find profile by id
+ * Note: This is called every time that the parameter :profileId is used in a URL. 
+ * Its purpose is to preload the profile on the req object then call the next function. 
+ */
+exports.getProfileId = function (req, res, next, id) {
```
Comment (jingwong25) [inline app/controllers/ProfileControllers.js]: getProfileById

**Comment by ykssheng** [comment]
please review again

**Comment by ykssheng** [comment]
Please retest and approve

**Comment by ykssheng** [comment]
{
    "firstName": "David",
    "lastName": "Tan",
    "address": null,
    "email": "cyew91@gmail.com",
    "contactNo": "12233221",
    "gender": "M",
    "dateOfBirth": "2018-09-09 05:13:25",
    "imageName": "image_A",
    "imagePath": "test_path",
    "loginId": "dtan",
    "password": "testing",
    "confirmPassword": "testing",
    "isActive": "1",
    "country_id": "005a1674-b3ef-11e8-a165-8086f20f4c17" -- follow ur own id
}

File: app/controllers/ProfileControllers.js [hunk 3]
```diff
@@ -178,4 +178,129 @@ exports.getProfileAccount = function (req, res, next, ProfileAccountID) {
...
+    });
+};
+
+// Update Profile
+exports.updateProfile = function (req, res) {
+
+    // create a new variable to hold the article that was placed on the req object.
+    var profile = req.profile;
+
+    profile.updateAttributes({
```
Comment (cyew91) [inline app/controllers/ProfileControllers.js]: Please include address, image, password and etc update also.

Comment (ykssheng) [reply app/controllers/ProfileControllers.js]: those fields will be added after that because is separated api

**Pull request merged**